### DISCUSSION TITLE
syntax error or bad assumptions?

### DISCUSSION THREAD

**Discussion by adamphetamine** [ariadar/Zoho-Creator-ChatGPT#2] [General]
![Screenshot 2023-04-25 at 7 06 20 pm](https://user-images.githubusercontent.com/47110465/234229240-e804630f-6eaf-47f1-b93e-fc8a78ff4bfb.png)
I'm getting an error with line 2-

if(input.Text_Input != "" && input.Refine_Input = false && input.Chat_Output == "")

because you're setting up a header map it shouldn't require a ';' at the end of the line, but perhaps I have done something else wrong?
I guess I should ask- what is needed to be set up before this will work?
Should it be a 'function' or a 'workflow'?

**Comment by ariadar** [comment]
A header map requires a ";" in the syntax. See [here](https://www.zoho.com/deluge/help/map-manipulations/create-map.html). 

The sample script is for deployment as a workflow. 

Often ZC will identify a problem citing a particular line number, but the bug could occur anywhere else in the logic that begins from that line number.

**Reply by adamphetamine** [reply]
thanks for the tip- I checked some other code that does not have the trailing ';'
but this gives me a few things to try, thanks again

**Comment by adamphetamine** [comment]
I have been able to get this code to work by simplifying it down to the first block and leaving out the conditional operands, and the first line of commenting- this makes it barf for some reason. 

However this line

`if(input.Text_Input != "" && input.Refine_Input = false && input.Chat_Output == "")`

Is still giving an error -

> 'In Criteria left expression is of type TEXT and right expression is of type
> TRUE/FALSE and the operator = is not valid'

So this must be something to do with the way you've set up the fields 
1. Text_Input
2. Refine_Input 
3. Chat_Output 

Can you supply any info about these please?
I've simply created 3x multi line fields with these names, and got the error above...

**Reply by ariadar** [reply]
Refine_Input, is a decision (true/false) button that refreshes your Text_Input, so you can use the davinci model to refine your question based on the previous Chat_Output.
[…](https://github.com/ariadar/Zoho-Creator-ChatGPT/discussions/2#)
